### ISSUE TITLE
Ported Macaroons.Net to run on .NET Core

### ISSUE THREAD

**Issue by cyberdot** [JornWildt/Macaroons.Net#8]
Hi Jorn, I have ported Macaroons.Net version for .NET Core as .NET Standard 2.0 library.
https://github.com/cyberdot/Macaroons.NETCore
Hope it's going to be useful for someone requiring Macaroons implementation for .NET Core (as I needed that as well)

**Comment by JornWildt** [comment]
That is neat :-) Are still using the same version of the crypto library? As I remember it, it is loaded from a C++ DLL ... how does that play with .NET Core?

**Comment by JornWildt** [comment]
Did you actually change any code - or was it just a matter of making a new csproj file for .NET Standard 2.0?

**Comment by cyberdot** [comment]
> That is neat :-) Are still using the same version of the crypto library? 


I had to use .NET Core specific packages: CuttingEdge.Conditions.NetStandard and Sodium.Core
Sodium.Core uses libsodium 1.0.18
Tested on Windows 10 and Linux Ubuntu 20.04

**Comment by cyberdot** [comment]
> Did you actually change any code - or was it just a matter of making a new csproj file for .NET Standard 2.0?

I had to change the code in a couple of places:
HexDataEncoding.cs - uses BitConverter instead of SoapHexBinary
SecretBoxCryptoAlgorithm.cs - Encrypt method, Sodium.SecretBox.Create no longer returns padded cipher text, had to update that as well
Utility.cs - uses memcmp method from msvcrt.dll, which is not portable, had to replace it with ReadOnlySpan<T>  implementation instead.

There is NuGet package available as well: https://www.nuget.org/packages/Macaroons.NETCore/

**Comment by JornWildt** [comment]
Nice work. I'm still somewhat new to .NET Core - but is it correct to assume that your .NET Core package should simply replace my .NET Standard solution? Your implementation should be plug-and-play replaceable for .NET Standard, right?

If so, then I won't make a NuGet package out of my version.

**Comment by JornWildt** [comment]
I see you still state me as the author on the NuGet package - not sure if that is right anymore.

Question ... should I pull your changes and release Macaroons.Net as a NuGet package from my repo, with your and my name on it - or would you rather just continue with your fork?

**Comment by cyberdot** [comment]
I'd be happy for you to incorporate my changes into your repo and have a single Macaroons.Net package supporting both .NET framework and .NET Core. Then I can remove my NuGet package from nuget.org

**Comment by JornWildt** [comment]
Cool. I'll see what I can do.

**Comment by JornWildt** [comment]
Turns out I don't know how to do it 🙄 According to Github there is no difference between your repo and my repo - so I cannot create a pull request (but, well, I haven't tried that before).

Can you make a pull request for me to pull into my repo?

**Comment by cyberdot** [comment]
Cool, no problem, I'll create a pull request

**Comment by JornWildt** [comment]
I am wondering if you have pushed your changes to Github? You have two visible commits - and neither of them includes any changes, only additions of files to the depot. If you haven't pushed the changes, it could explain why Git tells me that there are no changes to integrate from your repo.

**Comment by cyberdot** [comment]
I've just submitted a PR: https://github.com/JornWildt/Macaroons.Net/pull/9. 
- NuGet package meta information is residing in the project file now, this might affect the way you generate NuGet package.
- The original code for .NET framework is unchanged. The conditional changes are only to ensure it runs on .NET core
- I have used test project and console apps as a guidance to ensure correctness of the code changes
